### ISSUE TITLE
[Bug]: Update 27.1.1 fails with "Specified key was too long; max key length is 767 bytes"

### ISSUE THREAD

**Issue by Rudiberto** [nextcloud/server#40567]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

My update failed at the last stage with this error:

```
Update vorbereiten
Log-Level auf "debug" gesetzt
Wartungsmodus eingeschaltet
Reparaturschritt: Repair MySQL collation
Reparaturinformation: All tables already have the correct collation -> nothing to do
Reparaturschritt: Repair SQLite autoincrement
Reparaturschritt: Copy data from accounts table when migrating from ownCloud
Reparaturschritt: Drop account terms table when migrating from ownCloud
Das Datenbankschema wird aktualisiert

Exception: Database error when running migration 28000Date20230616104802 for app core An exception occurred while executing a query: SQLSTATE[42000]: Syntax error or access violation: 1071 Specified key was too long; max key length is 767 bytes

Exception: Database error when running migration 28000Date20230616104802 for app core An exception occurred while executing a query: SQLSTATE[42000]: Syntax error or access violation: 1071 Specified key was too long; max key length is 767 bytes
```

I updated from 27.0.2.1
The most important question is how to get back a working nextcloud.

### Steps to reproduce

1. made all steps 
2. failed in step 12 (yet with my main adress in the adress line)
3. I can try it endlessly, again and again

### Expected behavior

It should make the last step and then go into my Nextcloud installation.

### Installation method

Community Web installer on a VPS or web space

### Nextcloud Server version

27

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.1

### Web server

Apache (supported)

### Database engine version

MariaDB

### Is this bug present after an update or on a fresh install?

Updated from a MINOR version (ex. 22.1 to 22.2)

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [X] LDAP/ Active Directory
- [ ] SSO - SAML
- [X] Other

### Configuration report

```shell
That's not possible any more
```

### List of activated Apps

```shell
That's not possible any more
```

### Nextcloud Signing status

```shell
That's not possible any more
```

### Nextcloud Logs

```shell
Update vorbereiten
Log-Level auf "debug" gesetzt
Wartungsmodus eingeschaltet
Reparaturschritt: Repair MySQL collation
Reparaturinformation: All tables already have the correct collation -> nothing to do
Reparaturschritt: Repair SQLite autoincrement
Reparaturschritt: Copy data from accounts table when migrating from ownCloud
Reparaturschritt: Drop account terms table when migrating from ownCloud
Das Datenbankschema wird aktualisiert

Exception: Database error when running migration 28000Date20230616104802 for app core An exception occurred while executing a query: SQLSTATE[42000]: Syntax error or access violation: 1071 Specified key was too long; max key length is 767 bytes

more impossible for me. It's too huge to handle
```

**Comment by solracsf** [comment]
You checkmarked it, but [did you perform a search](https://github.com/search?q=repo%3Anextcloud%2Fserver+1071+Specified+key+was+too+long%3B+max+key+length+is+767+bytes&type=issues)? 🤔

**Comment by Rudiberto** [comment]
Yes I did, but I thought, that it could not be the "'mysql.utf8mb4' => false"-issue, because I worked the last six versions with 'mysql.utf8mb4' => "true". 

Thank you very much for the fingertip!!

So I can report now, that setting the value to "false" fixed the update error. After finishing the update with that value, I set the value back to true, and it ran through the whole process again (well only last step 12), but without the error I reported initially.
So at least I accomplished the update with your kindly help!
